smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/traits/EndpointTestsTraitValidator.java

```diff
@@ -28,6 +39,28 @@
 public final class EndpointTestsTraitValidator extends AbstractValidator {
     @Override
     public List<ValidationEvent> validate(Model model) {
-        return null;
+
+        List<ValidationEvent> result = new ArrayList<>();
+        for (ServiceShape service : model.getServiceShapesWithTrait(EndpointTestsTrait.class)) {
+            result.addAll(
```

Comment:
This would be a lot cleaner with iteration instead of streaming operations, removing three intermediate streams and leading to more readable code. The below method could also return a collection instead of a `Stream`.